### PR TITLE
Add --show-hidden flag to Deck

### PR DESCRIPTION
**Pull request by alvaroaleman**
This pull adds the `--show-hidden` flag to deck, allowing to run a Deck instance that shows both hidden and not-hidden jobs.

### CODE REVIEW

File: prow/kube/client.go [hunk 1]
```diff
@@ -75,17 +73,6 @@ type Client struct {
...
 	fake      bool
-
-	hiddenReposProvider func() []string
-	hiddenOnly          bool
-}
-
-// SetHiddenReposProvider takes a continuation that fetches a list of orgs and repos for
-// which PJs should not be returned.
-// NOTE: This function is not thread safe and should be called before the client is in use.
-func (c *Client) SetHiddenReposProvider(p func() []string, hiddenOnly bool) {
```
Comment (alvaroaleman) [inline prow/kube/client.go]: I've removed all of this because its unused. LMK if that is a problem

**Comment by alvaroaleman** [comment]
@Katharine Any chance you can have a look? :)

File: prow/cmd/deck/tide.go [hunk 2]
```diff
@@ -162,6 +163,9 @@ func (ta *tideAgent) filterHiddenPools(pools []tide.Pool) []tide.Pool {
 	filtered := make([]tide.Pool, 0, len(pools))
 	for _, pool := range pools {
 		needsHide := matches(pool.Org+"/"+pool.Repo, ta.hiddenRepos)
+		if needsHide && ta.showHidden {
+			filtered = append(filtered, pool)
+		}
 		if needsHide == ta.hiddenOnly {
```
Comment (Katharine) [inline prow/cmd/deck/tide.go]: Can this be `else if`? As it stands I can imagine future logic changes causing surprise duplicate entries.

Same goes for the two below.

Comment (alvaroaleman) [reply prow/cmd/deck/tide.go]: Done

File: prow/cmd/deck/main.go [hunk 3]
```diff
@@ -104,6 +105,10 @@ func (o *options) Validate() error {
 			return errors.New("an OAuth URL was provided but required flag --cookie-secret was unset")
 		}
 	}
+
+	if o.hiddenOnly && o.showHidden {
+		return errors.New("'--hidden-only' and '--show-hidden' are mutually exclusive")
```
[the remark this answers is not in the document]
Comment (alvaroaleman) [reply prow/cmd/deck/main.go]: Done

**Comment by alvaroaleman** [comment]
PTAL @Katharine

**Comment by fejta** [comment]
@Katharine @alvaroaleman not sure this PR warrants 7 commits? Heading into prod though now :)

**Pull request merged**